hs-ucan/library/Crypto/Key/Asymmetric/Public/Types.hs

```diff
@@ -61,12 +61,25 @@ instance IsString (Either Text Public) where
   fromString = parseUrlPiece . Text.pack
 
 instance FromJSON Public where
-  parseJSON =
-    withObject "PublicKey" \j ->
-      ((,) <$> j .: "type" <*> j .: "key") >>= \(t, k) ->
-      case fromKeyType t k of
-        Right pub -> return pub
-        Left err -> Json.parseFail (Text.unpack err)
+  parseJSON (Object j) =
+    ((,) <$> j .: "type" <*> j .: "key") >>= \(t, k) ->
+    case fromKeyType t k of
+      Right pub -> return pub
+      Left err -> Json.parseFail (Text.unpack err)
```

Comment:
Just saw this flying by and was thinking how about this?
```suggestion
  parseJSON (Object j) =
    (fromKeyType <$> j .: "type" <*> j .: "key") >>= \case
      Right pub -> return pub
      Left err -> Json.parseFail (Text.unpack err)
```

Or perhaps this is more readable:
```suggestion
  parseJSON (Object j) = do
    t <- j .: "type"
    k <- j .: "key"
    case fromKeyType t k of
      Right pub -> return pub
      Left err -> Json.parseFail (Text.unpack err)
```

Reply:
Thanks! Definitely more readable 😄